core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -80,10 +81,10 @@
 
     public ReplicationOperation(Request request, Primary<Request, ReplicaRequest, PrimaryResultT> primary,
                                 ActionListener<PrimaryResultT> listener,
-                                boolean executeOnReplicas, boolean checkWriteConsistency,
+                                boolean executeOnReplicas, boolean checkActiveShardCount,
                                 Replicas<ReplicaRequest> replicas,
                                 Supplier<ClusterState> clusterStateSupplier, ESLogger logger, String opType) {
-        this.checkWriteConsistency = checkWriteConsistency;
+        this.checkActiveShardCount = checkActiveShardCount;
```

Comment:
can we see check and see if we can get rid of this boolean infavour of setting the request waitForActiveShards to NONE by default on all the relevant requests?